packages/hardhat/contracts/Pools.sol

```diff
@@ -0,0 +1,161 @@
+pragma solidity ^0.8.4;
+//SPDX-License-Identifier: MIT
+
+import "hardhat/console.sol";
+import "@openzeppelin/contracts/token/ERC20/ERC20.sol";
+import "@openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol";
+import "prb-math/contracts/PRBMath.sol";
+import "prb-math/contracts/PRBMathUD60x18.sol";
+
+import "./interfaces/IPools.sol";
+import "./interfaces/Validated.sol";
+
+contract Pools is IPools, Validated {
+    using SafeERC20 for ERC20;
+    using PRBMathUD60x18 for uint256;
+
+    // This constant represents the utilization rate at which the pool aims to obtain most competitive borrow rates.
+    uint256 public immutable optimalUtilisationRate;
+    // This constant represents the excess utilization rate above the optimal.
+    // It's always equal to 1-optimal utilization rate. Added as a constant here for gas optimizations.
+    uint256 public immutable excessUtilisationRate;
+    // Base variable borrow rate when Utilization rate = 0.
+    uint256 internal immutable baseBorrowRate;
+    // Slope of the variable interest curve when utilization rate > 0 and <= optimalUtilisationRate.
+    uint256 internal immutable slope1;
+    // Slope of the variable interest curve when utilization rate > optimalUtilisationRate.
+    uint256 internal immutable slope2;
```

Comment:
This config is on a per-ERC20 basis

Reply:
Probably better to use a different struct so we don't read them all the time for no reason